### DISCUSSION TITLE
Real Time Lighting

### DISCUSSION THREAD

**Discussion by ash-ashash-ash** [ikemen-engine/Ikemen-GO#908] [Feature Request]
With the ability to use ReShade with Ikemen GO, it opens up many possibilities for stages to truly look amazing, especially those with lighting; However, the issue is that this lighting is achieved using gradients and otherwise tinting the scene. A way to actually make ReShade work perfect with Ikemen GO would be to add a true lighting feature for stages and perhaps characters, thus allowing for more complex shaders. I truly understand if this would be difficult to implement in a 2-D scene.

Include:
 - Base lighting controller (defaults to fullbright)
 - Lighting controller, supports different types of light through a lighting type option

TL;DR: Real time lighting in Ikemen would be boss.
